File: example/src/com/appbrain/example/ListAdsActivity.java
package com.appbrain.example;

import android.app.ListActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.AdapterView.OnItemClickListener;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

import com.appbrain.AdListAdapter;
import com.appbrain.AppBrain;

public class ListAdsActivity extends ListActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ListView listView = getListView();

        ArrayAdapter<String> adapter = new ArrayAdapter<String>(this,
            android.R.layout.simple_list_item_1);
        for (int i = 0; i < 100; i++) {
            adapter.add("Item " + i);
        }

        final AdListAdapter adAdapter = AppBrain.getAds().wrapListAdapter(this, adapter);
        listView.setAdapter(adAdapter);
        listView.setOnItemClickListener(new OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                int itemPosition = adAdapter.getItemPosition(position);
                Toast.makeText(ListAdsActivity.this, "Item " + itemPosition + " pressed",
                    Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: example-compose/src/com/appbrain/example/ListAdsActivity.kt
package com.appbrain.example

import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.ListView
import androidx.activity.ComponentActivity
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.viewinterop.AndroidView

import com.appbrain.AppBrain

class ListAdsActivity : ComponentActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setExampleContent {
            ListAdsScreen()
        }
    }

    @Composable
    private fun ListAdsScreen() {
        // wrapListAdapter works with the classic ListView/ListAdapter, so we host
        // a ListView through Compose's AndroidView interop.
        AndroidView(
            modifier = Modifier.fillMaxSize(),
            factory = { ctx ->
                val adapter = ArrayAdapter<String>(ctx, android.R.layout.simple_list_item_1)
                for (i in 0 until 100) {
                    adapter.add("Item $i")
                }

                val adAdapter = AppBrain.getAds().wrapListAdapter(ctx, adapter)
                ListView(ctx).apply {
                    setAdapter(adAdapter)
                    setOnItemClickListener { _, _, position, _ ->
                        val itemPosition = adAdapter.getItemPosition(position)
                        ctx.toast("Item $itemPosition pressed")
                    }
                }
            }
        )
    }
}
